subprojects/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/AbstractJavaGroovyIncrementalCompilationSupport.groovy

```diff
@@ -53,7 +53,7 @@ abstract class AbstractJavaGroovyIncrementalCompilationSupport extends AbstractI
             def packageGroup = (body =~ "\\s*package ([\\w.]+).*")
             String packageName = packageGroup.size() > 0 ? packageGroup[0][1] : ""
             String packageFolder = packageName.replaceAll("[.]", "/")
-            def className = (body =~ /(?s).*?(?:class|interface|enum) ([\w$]+) .*/)[0][1]
+            def className = (body =~ /(?s).*?(?:class|interface|enum|trait) ([\w$]+) .*/)[0][1]
```

Comment:
❓ How is this change related to this PR?

Reply:
This code creates a file directly from string source.

In the test we added a source:
```
trait WithTrait {
...
}
```
If we want that `WithTrait.groovy` is correctly created, we also have to parse `className` where class starts with `trait <ClassName>`.